opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,172 @@
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
+                _logger.warning(
+                    "Instrument type matching disabled "
+                    "for view with a defined name"
+                )
+                self._instrument_type = None
+
+            if self._instrument_name is not None:
+                _logger.warning(
+                    "Instrument name matching disabled "
+                    "for view with a defined name"
+                )
+                self._instrument_name = None
+
+            if self._meter_name is not None:
+                _logger.warning(
+                    "Instrument meter name matching disabled "
+                    "for view with a defined name"
+                )
+                self._meter_name = None
+
+            if self._meter_version is not None:
+                _logger.warning(
+                    "Instrument meter version matching disabled "
+                    "for view with a defined version"
+                )
+                self._meter_version = None
+
+            if self._meter_schema_url is not None:
+                _logger.warning(
+                    "Instrument meter schema URL matching disabled "
+                    "for view with a defined version"
+                )
+                self._meter_schema_url = None
+
+        self._description = description
+        self._attribute_keys = attribute_keys
+        self._aggregation = aggregation
+
+        self._has_name_and_has_matched = False
+
+    # pylint: disable=too-many-return-statements
+    def _match(self, instrument: Instrument) -> bool:
+
+        if self._has_name_and_has_matched:
+            return False
+
+        if self._instrument_type is not None:
+            if not isinstance(instrument, self._instrument_type):
+                return False
+
+        if self._instrument_name is not None:
+            if match(self._instrument_name, instrument.name) is None:
+                return False
+
+        if self._meter_name is not None:
+            # pylint: disable=protected-access
+            if self._meter_name != instrument._meter.name:
+                return False
+
+        if self._meter_version is not None:
+            # pylint: disable=protected-access
+            if self._meter_version != instrument._meter.version:
+                return False
+
+        if self._meter_schema_url is not None:
+            # pylint: disable=protected-access
+            if self._meter_schema_url != instrument._meter.schema_url:
+                return False
+
+        if self._name is not None:
+            self._has_name_and_has_matched = True
```

Comment:
Curious as to why we are tracking that the name has to be defined?

Reply:
Per spec:
> In order to avoid conflicts, views which specify a name SHOULD have an instrument selector that selects at most one instrument.

Reply:
I think this is related to my other question so we can discuss in the other conversation.